client/SFTPClientGUI/SFTPClientGUI.py

```diff
@@ -0,0 +1,366 @@
+		self.treeview_local.connect('row-collapsed', self.signal_collapse_row)
+		self._tvmodel = Gtk.TreeStore(str, Pixbuf, str, str, str, int, str)
+
+		self.local_hidden = True
+		self._get_popup_menu()
+
+	def _get_popup_menu(self):
+		self.treeview_local.connect('button_press_event', self._signal_treeview_button_pressed)
+
+		self.popup_menu = Gtk.Menu.new()
+		menu_item = Gtk.CheckMenuItem.new_with_label('Show Hidden Files')
+		menu_item.connect('toggled', self.signal_menu_toggled_hidden_files)
+		self.signal_menu_toggled_hidden_files = menu_item
+		self.popup_menu.append(menu_item)
+
+		menu_item = Gtk.MenuItem.new_with_label('Delete')
+		menu_item.connect('activate', self.signal_menu_activate_delete_prompt)
+		self.popup_menu.append(menu_item)
+		self.popup_menu.show_all()
+
+	def _signal_treeview_button_pressed(self, _, event):
+		if event.button == Gdk.BUTTON_SECONDARY:
+			self.popup_menu.popup(None, None, None, None, event.button, Gtk.get_current_event_time())
+			return True
+		return
+
+	def signal_expand_row(self, _, treeiter, treepath):
+		new_path = self._tvmodel[treeiter][2]  # pylint: disable=unsubscriptable-object
+		self.load_dirs(new_path, treeiter)
+		self._tvmodel.remove(self._tvmodel.iter_children(treeiter))
+
+	def signal_collapse_row(self, _, treeiter, treepath):
+		current = self._tvmodel.iter_children(treeiter)
+		while current:
+			self._tvmodel.remove(current)
+			current = self._tvmodel.iter_children(treeiter)
+		self._tvmodel.append(treeiter, [None, None, None, None, None, None, None])
+
+	def load_dirs(self, path, parent=None):
+		counter = 0
+		for name in self._yield_dir_list(path):
+			if self.local_hidden and name.startswith('.'):
+				continue
+			fullname = path + '/' + name
+			perm = self._check_perm(fullname)
+			raw_time = self._get_raw_time(fullname)
+			date_modified = '   ' + time.strftime('%y-%m-%d %H:%M:%S', time.localtime(raw_time))
+			is_folder = self._get_is_folder(fullname)
+			if is_folder:
+				icon = Gtk.IconTheme.get_default().load_icon('folder', 20, 0)
+				if perm[3] != 'r' and perm[5] != 'x':
+					icon = Gtk.IconTheme.get_default().load_icon('emblem-unreadable', 13, 0)
+				current = self._tvmodel.append(parent, (name, icon, fullname, perm, None, -1, date_modified))
+			else:
+				file_size = self._get_file_size(fullname)
+				hr_file_size = '   ' + boltons.strutils.bytes2human(file_size)
+				if perm[3] != 'r' and perm[5] != 'x':
+					icon = Gtk.IconTheme.get_default().load_icon('emblem-unreadable', 13, 0)
+				icon = Gtk.IconTheme.get_default().load_icon('empty', 12.5, 0)
+				current = self._tvmodel.append(parent, (name, icon, fullname, perm, hr_file_size, file_size, date_modified))
+			if is_folder and perm[3] == 'r' and perm[5] == 'x':
+				self._tvmodel.append(current, [None, None, None, None, None, None, None])
+			counter += 1
+		if counter < 1:
+			self._tvmodel.append(parent, [None, None, None, None, None, None, None])
```

Comment:
I believe this is making a blank line show up when an empty directory is expanded. What would be ideal is when an empty directory is expanded if the arrow allowing the directory to be expanded would be removed after it's know to be empty.